packages/agreement-process/src/services/agreementService.ts

```diff
@@ -468,7 +471,7 @@ export async function createAgreementLogic(
 
   const descriptor = validateCreationOnDescriptor(
     eservice.data,
-    agreement.descriptorId
+    unsafeBrandId(agreement.descriptorId)
```

Comment:
Isn't this already branded?

Reply:
sadly no! In this case, the `agreement` var comes from zodios decoder. Right now we don't have a way to add a `brand` decoder on these fields. My PR to add this functionality upstream has been rejected and now I'm waiting for another PR that could implement a feature to resolve this issue.